frontend/pages/js/execution_logic.js

```diff
@@ -0,0 +1,285 @@
+
+      case CODE_STATUS.NOT_CONNECTED:
+        launch_button.innerHTML = "Server is down";
+        break;
+
+      default:
+        console.log('Unknown current status: ' + current_status);
+        break;
+    }
+  }
+
+  function update_workspace() {
+    var blocks_tab_selector = "a[href='#home'][data-toggle='tab']";
+    var python_tab_selector = "a[href='#profile'][data-toggle='tab']";
+    var graph_tab_selector = "a[href='graph.html']";
+    var load_from_file_button_selector = "a[id='load_from_file_button']";
+    var save_to_file_button_selector = "a[id='save_to_file_button']";
+    switch (current_status) {
+      case CODE_STATUS.PAUSED:
+      case CODE_STATUS.RUNNING:
+        workspace.options.readOnly = true;
+        if (null != workspace.toolbox_) {
+          workspace.toolbox_.HtmlDiv.hidden = true;
+        }
+        $(blocks_tab_selector).hide();
+        $(python_tab_selector).hide();
+        $(graph_tab_selector).hide();
+        $(load_from_file_button_selector).hide();
+        $(save_to_file_button_selector).hide();
+        break;
+
+      case CODE_STATUS.COMPLETED:
+      case CODE_STATUS.NOT_CONNECTED:
+        current_block = null;
+        var blocks = workspace.getAllBlocks();
+        for (var i = 0; i < blocks.length; i++) {
+          blocks[i].setShadow(false);
+        }
+        workspace.options.readOnly = false;
+        if (null != workspace.toolbox_) {
+          workspace.toolbox_.HtmlDiv.hidden = false;
+        }
+        $(blocks_tab_selector).show();
+        $(python_tab_selector).show();
+        $(graph_tab_selector).show();
+        $(load_from_file_button_selector).show();
+        $(save_to_file_button_selector).show();
+        break;
+
+      default:
+        console.log('Unknown current status: ' + current_status);
+        break;
+    }
+  }
+
+  function set_current_status(value) {
+    var statuses = [CODE_STATUS.COMPLETED, CODE_STATUS.RUNNING, CODE_STATUS.PAUSED, CODE_STATUS.NOT_CONNECTED];
+    if (0 > statuses.indexOf(value)) {
+      console.log('Unknown status: ' + value);
+      return;
+    }
+    current_status = value;
+    update_launch_button();
+    update_workspace();
+  }
+
+  function set_current_block_id(block_id) {
+    if ([CODE_STATUS.RUNNING, CODE_STATUS.PAUSED].indexOf(current_status) >= 0) {
+      var selected_block = workspace.getBlockById(block_id);
+      if (null != selected_block) {
+        if (null != current_block) {
+          current_block.setShadow(false);
+        }
+        selected_block.setShadow(true);
+        current_block = selected_block;
+      } else {
+        console.log('Not existing block id: ' + block_id);
+      }
+    } else {
+      console.log('Code is not running. Ignoring current block changed event.')
+    }
+  }
+
+  function is_connection_closed() {
+    return (CODE_STATUS.NOT_CONNECTED == current_status);
+  }
+
+  return {
+
+    launch_websockets: function () {
+      socket = new WebSocket("ws://0.0.0.0:9000");
```

Comment:
please change this line to:
`
var hostName = window.location.hostname;
socket = new WebSocket("ws://" + hostName + ":9000");
`